pytorch_toolkit/formula_recognition/README.md

```diff
@@ -0,0 +1,199 @@
+# PyTorch realization of the Formula Recognition
+
+This code is based on a [PyTorch realization](https://github.com/luopeixiang/im2latex/) of the code from the original [repository](https://github.com/harvardnlp/im2markup/).
+Models code is designed to enable ONNX\* export and inference on CPU\GPU via OpenVINO™.
+
+## Setup
+
+### Prerequisites
+
+* Ubuntu\* 16.04
+* Python\* 3.7 or newer
+* PyTorch\* (1.4.0)
+* OpenVINO™ 2020.4 with Python API
+
+### Optional prerequisites
+
+#### Install required packages for evaluation
+These packages are used for rendering images while evaluation and demo.
+
+```bash
+sudo apt-get update &&
+  apt-get install -y --no-install-recommends \
+    texlive \
+    imagemagick \
+    ghostscript
+```
+
+### Installation
+
+Create and activate virtual environment:
+
+```bash
+bash init_venv.sh
+```
+
+### Download or Prepare Datasets
+
+Dataset format is similar to [im2latex-100k](https://zenodo.org/record/56198#.X2NDQ2gzaUl). Main structure of the dataset is following:
+* `formulas_file` - file with one formula per line
+* `images_folder` - folder containing input images
+* `split_file` - this file contains `image_name` (tab symbol) `formula_idx` per line connecting corresponding index of the formula in the file with formulas and particular image with `image_name`. Example:
+    ```
+    11.png  11
+    34.png  34
+    ...
+    ```
+    There should be at least two such files: `train_filter.lst` and `validate_filter.lst`
+
+You can prepare your own dataset in the same format as above. Samples of the dataset can be found [here](../../data/formula_recognition). When you prepare your own dataset with `formulas.norm.lst` file, you will have to create a vocabulary file for this dataset. Vocabulary file is a special file which is used to cast token ids to human readable tokens and vice versa. Like letters and digits in the natural language, tokens here are atomic units of the latex language (e.g. `\\sin`, `1`, `\\sqrt`, etc). You can find an example in the [vocabs folder](./vocabs/) of this project. Use [this script](./tools/make_vocab.py) to create vocab file from your own formulas file. The script will read the formulas and create the vocabulary from the formulas used in train split of the dataset.
+
+> **NOTE**:
+> By default the following structure of the dataset is assumed:
+> `images_processed` - folder with images
+> `formulas.norm.lst` - file with preprocessed formulas. If you want to use your own dataset, formulas should be preprocessed. For details, refer to [this script](https://github.com/harvardnlp/im2markup/blob/master/scripts/preprocessing/preprocess_formulas.py).
+> `validate_filter.lst` and `train_filter.lst` - corresponding splits of the data.
```

Comment:
I'd propose to move this whole note above, just before the subsection `#### Vocabulary file`